Refactor Documentation (#645)
graalvm/native-build-tools: docs/src/docs/asciidoc/index.adoc

= Native Build Tools
The GraalVM team

> The {doctitle} project provides plugins for different build tools to add support for building and testing native applications written in Java

Reviewer: I didn't hear about "native projects", but "native applications"

(or any other language compiled to JVM bytecode) using the https://www.graalvm.org/reference-manual/native-image/[GraalVM  Native Image].
At the moment, you can build your native projects using our Gradle or Maven plugin.

Each plugin takes metadata for available libraries from the https://github.com/oracle/graalvm-reachability-metadata[GraalVM reachability metadata repository]
to provide better experience during the development of your native project.

If you are using alternative build systems, see <<alternative-build-systems.adoc#,Useful Hints for Alternative Build Systems>>.

[discrete]
=== Documentation

> In case you are new to the Native Build Tools (either as an end user or a library author), read our:

Reviewer: ```suggestion
If you're new to the Native Build Tools, whether as an end user or a library author, check out our:
```

> - <<end-to-end-gradle-guide.adoc#,End-to-end Gradle guide>>

Reviewer: also we should change the file names accordingly

> - <<end-to-end-gradle-guide.adoc#,End-to-end Gradle guide>>

Author: Addressed [here](https://github.com/graalvm/native-build-tools/pull/645#discussion_r1881964618)

- <<end-to-end-maven-guide.adoc#,End-to-end Maven guide>>

For detailed documentation, please refer to the following pages of the specific build tools:

- <<gradle-plugin.adoc#,The Gradle plugin documentation>>
- <<maven-plugin.adoc#,The Maven plugin documentation>>

You can see the full changelog <<changelog.adoc#,here>>.

[discrete]
=== Contributing

If you are interested in contributing or reporting an issue related to the plugins, please refer to the https://github.com/graalvm/native-build-tools[Native Build Tools repository].

> Also, be aware that all contributors must sign the https://oca.opensource.oracle.com/[Oracle Contributor Agreement] (OCA).

Reviewer: ```suggestion
```suggestion
Please note that all contributors must sign the https://oca.opensource.oracle.com/[Oracle Contributor Agreement] (OCA).
```
